rllib/algorithms/ppo/torch/ppo_torch_learner.py

```diff
@@ -47,10 +47,11 @@ def compute_loss_for_module(
         # for which we add an additional (artificial) timestep to each episode to
         # simplify the actual computation.
         if Columns.LOSS_MASK in batch:
-            num_valid = torch.sum(batch[Columns.LOSS_MASK])
+            mask = batch[Columns.LOSS_MASK]
+            num_valid = torch.sum(mask)
 
             def possibly_masked_mean(data_):
-                return torch.sum(data_[batch[Columns.LOSS_MASK]]) / num_valid
+                return torch.sum(data_[mask]) / num_valid
```

Comment:
Could we potentially also mask with this `mask` values in the observations that are not available in a certain step of the environment, e.g. different number of entities at different steps of a game?

Reply:
That should work. But you would have to manually change that in the Learner connector (where this column is being produced and added to the batch).